src/HDWallet.cpp

```diff
@@ -110,6 +110,12 @@ PrivateKey HDWallet::getKey(TWCoinType coin, TWDerivation derivation) const {
     return getKey(coin, path);
 }
 
+PrivateKey HDWallet::getCurveKey(TWCurve curve, const DerivationPath& derivationPath) const {
+    auto node = getNode(*this, curve, derivationPath);
+    auto data = Data(node.private_key, node.private_key + PrivateKey::size);
+    return PrivateKey(data);
+}
+
```

Comment:
This should handle some other private key types as well.
My suggestion:
- Adapt existing getKey() to take curve instead of coin (omit first line to get curve from coin), rename to getKeyByCurve.
- Add new getKey() to invoke it 

```
PrivateKey HDWallet::getKey(TWCoinType coin, const DerivationPath& derivationPath) const {
    const auto curve = TWCoinTypeCurve(coin);
    return getKeyByCurve(curve, derivationPath);
}
```

Reply:
The purpose of the function is to handle some of the blockchain (eg. [Flow](https://docs.onflow.org/cadence/language/crypto/#signing-algorithms)) allow multiple curve. For now, I can see the current the protocol in `CoinType` only return one curve in `TWCoinTypeCurve` func. Hence, as a minimum change here. 
Hence I think add the `getKeyByCurve` will be useful in this stage. Since It's more flexible.